include/boost/gil/image_processing/filter.hpp

```diff
@@ -0,0 +1,64 @@
+//
+// Copyright 2019 Miral Shah <miralshah2211@gmail.com>
+//
+// Use, modification and distribution are subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef BOOST_GIL_IMAGE_PROCESSING_FILTER_HPP
+#define BOOST_GIL_IMAGE_PROCESSING_FILTER_HPP
+
+#include <cstddef>
+#include <vector>
+
+#include <boost/gil/image.hpp>
+#include <boost/gil/image_view.hpp>
+#include <boost/gil/extension/numeric/kernel.hpp>
+#include <boost/gil/extension/numeric/convolve.hpp>
+
+namespace boost { namespace gil {
+
+template <typename SrcView, typename DstView>
+void box_filter(
+    SrcView const& src_view,
+    DstView const& dst_view,
+    std::size_t kernel_size,
+    long int anchor = -1,
+    bool normalize=true,
+    convolve_boundary_option option = convolve_option_extend_zero
+)
+{
+    gil_function_requires<ImageViewConcept<SrcView>>();
+    gil_function_requires<MutableImageViewConcept<DstView>>();
+    static_assert(color_spaces_are_compatible
+    <
+        typename color_space_type<SrcView>::type,
+        typename color_space_type<DstView>::type
+    >::value, "Source and destination views must have pixels with the same color space");
+
+    std::vector<float> kernel_values;
+    if (normalize) { kernel_values.resize(kernel_size, 1.0f / float(kernel_size)); }
+    else { kernel_values.resize(kernel_size, 1.0f); }
+
+    if (anchor == -1) anchor = kernel_size / 2;
+    kernel_1d<float> kernel(kernel_values.begin(), kernel_size, anchor);
+
+    convolve_1d<pixel<float, typename SrcView::value_type::layout_t>>(src_view, kernel, dst_view);
+}
+
+template <typename SrcView, typename DstView>
+void blur(
+    SrcView const& src_view,
+    DstView const& dst_view,
+    std::size_t kernel_size,
+    std::size_t anchor = kernel_size / 2,
```

Comment:
@miralshah365 Such re-use of arguments is not valid C++ as per http://eel.is/c++draft/dcl.fct.default#9

> A default argument is evaluated each time the function is called with no argument for the corresponding parameter.
> A parameter shall not appear as a potentially-evaluated expression in a default argument.
> Parameters of a function declared before a default argument are in scope and can hide namespace and class member names.
> 
> ```
> int a;
> int f(int a, int b = a);            // error: parameter a used as default argument
> ```